Lib9c/Action/BattleArena.cs

```diff
@@ -248,9 +249,17 @@ public override IAccountStateDelta Execute(IActionContext context)
             var interval = gameConfigState.DailyArenaInterval;
             var currentTicketResetCount = ArenaHelper.GetCurrentTicketResetCount(
                 context.BlockIndex, roundData.StartBlockIndex, interval);
+            var purchasedCountAddr = arenaInformation.Address.Derive(PurchasedCountKey);
+            if (!states.TryGetState(purchasedCountAddr, out Integer purchasedCountDuringInterval))
+            {
+                purchasedCountDuringInterval = 0;
+            }
+
             if (arenaInformation.TicketResetCount < currentTicketResetCount)
             {
                 arenaInformation.ResetTicket(currentTicketResetCount);
+                purchasedCountDuringInterval = 0;
+                states = states.SetState(purchasedCountAddr, purchasedCountDuringInterval);
```

Comment:
Isn't interval should be keeping count throughout the season? In this case, it seems that the interval is always reset to 0 every cycle.

Reply:
The intention is correct!